src/dotty/tools/dotc/core/tasty/NameBuffer.scala

```diff
@@ -40,13 +41,12 @@ class NameBuffer extends TastyBuffer(10000) {
       nameIndex(name)
   }
 
-  private def withLength(op: => Unit): Unit = {
+  private def withLength(op: => Unit, lengthWidth: Int = 1): Unit = {
     val lengthAddr = currentAddr
-    writeByte(0)
+    for (i <- 0 until lengthWidth) writeByte(0)
     op
     val length = currentAddr.index - lengthAddr.index - 1
-    assert(length < 128)
-    putNat(lengthAddr, length, 1)
+    putNat(lengthAddr, length, lengthWidth)
```

Comment:
My only question is: what happens when the (altered) assertion fails? There can't be bounds-checking, so `assert(length < (1 << (7 * lengthWidth)))` would be good. I assume you're sure this will never happen, and that you're currently right, that's why I propose an assert, which would trigger if `NameBuffer` gets out-of-sync and `lengthWidth` is 1 when it shouldn't. Or simply if somebody forgets to specify `lengthWidth` 😉.